### ISSUE TITLE
A11y_SQLMigrationControlPlane_ AzureSQLMigration _MigrationStatus_ScreenReader: Buttons present inside the table is announced as group by the screen reader.

### ISSUE THREAD

**Issue by BhanuPrakashGit** [microsoft/azuredatastudio#20632]
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags:#A11yMAS;#A11yTCS;#AzureDataStudioextensionAzureSQLMigration-Mac-Sept2022;#SQL Migration Control Plane;#A11ySev2;#WCAG1.3.1;#DesktopApp;#Win11;#ScreenReader;#Benchmark;#VoiceOver;#Mac

### Environment Details:
Azure Data Studio: Version: 1.39.1
Operating System: Mac Version 12.5.1
Screen reader: VoiceOver

### Steps to Reproduce:

1. Install Azure data studio and open
2. Tab till Extension press enter key and install Azure SQL Migration
3. Tab till Respective server and right click
4. Tab till manage and press enter key
5. Tab till Azure SQL Migration and press enter key
6. Tab till migrate to azure sql and press enter key
7. Saved session page will open
8. Select start new session and give next
9. Databases for assessment page will open
10. Select the respective databases and give next
11. Assessment results and recommendations page will open
12. Select Azure SQL managed Instance /Virtual Machinesfrom Azure SQL Target
13. Tab till View/Select button, assessment results for '' dialog will open
14. Observe that keyboard focus is going over the non-interactive text.
15. Open Screen reader and observe that the buttons present on the table are announced as the group.

### Actual:
Buttons present inside the table is announced as group by the screen reader.

### Expected:
Buttons present inside the table should announced as button by the screen reader.

### User Impact:
Screen reader users will get confused with the controls present on the screen.
[Screen Recording 2022-09-15 at 10.46.47 AM.zip](https://github.com/microsoft/azuredatastudio/files/9616576/Screen.Recording.2022-09-15.at.10.46.47.AM.zip)

**Comment by InduPriya1805** [comment]
Githubtags:  #AzureDataStudioextensionAzureSQLMigration-Mac-Sept2022;

**Comment by brian-harris** [comment]
@aasimkhan30 is this issue fixed now?

**Comment by aasimkhan30** [comment]
@brian-harris, yup it is fixed.

**Comment by brian-harris** [comment]
@BhanuPrakashGit, closing this bug as it should now be fixed, please reopen if not.

**Comment by BhanuPrakashGit** [comment]
Hi @brian-harris tried to verify the bug in the environment: 
Version: 1.47.0-insider (Universal)
Commit: a63463e35bb21bb89250be09d9639ff440c83b39
Date: 2023-09-21T05:42:38.696Z
VS Code: 1.79.2
Electron: 22.3.14
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Darwin arm64 22.6.0

but getting below error, could you please look into this.

![image](https://github.com/microsoft/azuredatastudio/assets/104351566/8713bbfb-b623-48e2-8070-2cbb30937d28)

**Comment by BhanuPrakashGit** [comment]
Hi @brian-harris, I downloaded the Azure Data Studio from the below link,
https://github.com/microsoft/azuredatastudio#try-out-the-latest-insiders-build-from-main-branch

I got the same version as mentioned above and getting same error. Could you please look into this.

**Comment by brian-harris** [comment]
@BhanuPrakashGit , my team has created this suggestion to a similar issue.  Can you see if this method works for you?

- Azure Data Studio issues:

- Failed to start Sql Migration Service: Error: Request error:
Message: Error at ClientRequest.<anonymous> (c:\Users\et56892.azuredatastudio\extensions\microsoft.sql-migration-1.4.2\dist\main.js:2:7448) at ClientRequest.emit (node:events:538:35) at TLSSocket.socketOnEnd (node:_http_client:466:9) at TLSSocket.emit (node:events:538:35) at endReadableNT (node:internal/streams/readable:1345:12) at process.processTicksAndRejections (node:internal/process/task_queues:83:21)

- Cause: This happens when ADS is not able to download the MigrationService package from https://github.com/microsoft/sqltoolsservice/releases . The download failure can be due to disconnected network work or unresolved proxy settings. We are working on a fix.
Recommendation : 

- The surefire way of solving this is by downloading the package manually. Follow the mitigation steps outlined in this link: https://github.com/microsoft/azuredatastudio/issues/22558#issuecomment-1496307891

**Comment by BhanuPrakashGit** [comment]
Please do not close this bug. This bug should only be closed by TCS, C+AI Accessibility tester after bug verification.

Hi @brian-harris Tried the steps shared above but steps are not clear as in the first steps, it says to run command but terminal shows, there is no such command.
Could you please share the clear steps so that I can verify this issue
![image](https://github.com/microsoft/azuredatastudio/assets/104351566/19d5e012-d6e4-4c9a-a65f-995964695a30)

<img width="782" alt="MicrosoftTeams-image (30)" src="https://github.com/microsoft/azuredatastudio/assets/104351566/f20c00b7-0b85-4f0f-8fa0-108bb733094b">

**Comment by brian-harris** [comment]
re-opening bug as requested:
Does this work for you?
Open command pallet
![image](https://github.com/microsoft/azuredatastudio/assets/61598682/972f923d-fc59-4234-87ea-56c2e1dd3e50)

Open extensions directory
![image](https://github.com/microsoft/azuredatastudio/assets/61598682/11aa5731-31bf-42a9-9e6c-0d064a61fa3f)

Find sql-migration
![image](https://github.com/microsoft/azuredatastudio/assets/61598682/aa5e5348-d061-4276-bd48-4b8e3d5d8809)

**Comment by BhanuPrakashGit** [comment]
@brian-harris Yes, I'm able find this folder. What's the next steps? Please let me know.

**Comment by brian-harris** [comment]
please continue to follow these instructions:
![image](https://github.com/microsoft/azuredatastudio/assets/61598682/fbfb5367-d7be-4dfb-bf53-0fa24d9955f6)

**Comment by BhanuPrakashGit** [comment]
Hi @brian-harris I tried to verify the issue but I'm not able to require subscription name because of that I'm not able to proceed further.
Could you please help me here

![image](https://github.com/microsoft/azuredatastudio/assets/104351566/8e3c9f22-4246-419a-b1db-a68633c1345c)

**Comment by brian-harris** [comment]
@BhanuPrakashGit can you please verify you have a SQL Managed Instance created in the One Integration Service subscription.  The target selection page will only show locations and resource groups that contain a SQL Managed Instance.  If there aren't any in the subscription, it will be blank.

**Comment by brian-harris** [comment]
This tutorial may be helpful:  https://learn.microsoft.com/en-us/azure/dms/tutorial-sql-server-managed-instance-online-ads

**Comment by BhanuPrakashGit** [comment]
@brian-harris Earlier I got the access for the below subscription, but this is not available in the subscription field now.

![image](https://github.com/microsoft/azuredatastudio/assets/104351566/3cd446e3-d1d4-4d5b-8ac8-bbf36fe32cdc)

**Comment by brian-harris** [comment]
@BhanuPrakashGit, you should now have access to the subscription, please try again.

**Comment by BhanuPrakashGit** [comment]
@brian-harris Tried to Verify in the server but getting below error, Could you please look into this. Seems provided server is not working anymore.

<img width="1440" alt="NoAccess" src="https://github.com/microsoft/azuredatastudio/assets/104351566/370bc646-444e-4987-97ec-3c8e55a1fa42">
<img width="1440" alt="No access" src="https://github.com/microsoft/azuredatastudio/assets/104351566/77eb6213-426b-4e1f-b419-4a702108c09c">

**Comment by BhanuPrakashGit** [comment]
Hi @brian-harris Below is the proxy configuration setting in ADS.

<img width="1440" alt="MicrosoftTeams-image (32)" src="https://github.com/microsoft/azuredatastudio/assets/104351566/79056c93-334a-496b-944d-4e4fd1690940">

**Comment by BhanuPrakashGit** [comment]
@brian-harris Could you please confirm whether the UI for Migration status is changed, please find the screen shot below

![MicrosoftTeams-image (34)](https://github.com/microsoft/azuredatastudio/assets/104351566/159db866-166b-4a16-a397-905f51fdccc1)

is this the latest screen for Migration status? Please confirm on this

**Comment by brian-harris** [comment]
This looks like the latest.  Please ensure you have the latest update ADS and the SQL Migration extension:
![image](https://github.com/microsoft/azuredatastudio/assets/61598682/599f48d5-76db-4c3c-83c2-00a865bfa4b8)

**Comment by BhanuPrakashGit** [comment]
@brian-harris Tried to verify the bug in the environment:
Version: 1.48.0-insider (Universal)
Commit: e5e1204d4fb4d4e880cdd893347752ee5736459a
Date: 2023-12-05T06:38:01.191Z
VS Code: 1.82.0
Electron: 25.9.6
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.1.0

I'm unable to verify the issue as screen reader focus is not going over the buttons 'DWQueue' and 'Restoring'. 
Could you please look into this.
Please find the attachment

[20632_Not Fixed.zip](https://github.com/microsoft/azuredatastudio/files/13579442/20632_Not.Fixed.zip)

**Comment by cpichuka** [comment]
Tagging @krritik for this bug.

**Comment by krritik** [comment]
Focus of sequential keyboard navigation (e.g., pressing the "Tab" key) is not going over the hyperlinks in the table component row elements.


Issue is that tabIndex  is set as -1 by default for hyperlink elements in table component. It should be 0, then only sequential focus will work. We are using `azdata.TableComponent` for table and for hyperlink table components, we are using `azdata.HyperlinkColumnCellValue`.

![image](https://github.com/microsoft/azuredatastudio/assets/32814075/64ea6765-56d8-46ff-aed1-a9667956aef9)

The bug is on ADS core team. 
@alanrenmsft @aasimkhan30

**Comment by cpichuka** [comment]
@BhanuPrakashGit , did you log bug for ADS?

**Comment by BhanuPrakashGit** [comment]
Githubtags:#Regressed:12-27-23;#External;#External:ADS;#FirstParty:Bug25150
Logged External bug: https://github.com/microsoft/azuredatastudio/issues/25150#issue-2057171925

**Comment by alanrenmsft** [comment]
@kisantia fyi.

**Comment by kisantia** [comment]
@BhanuPrakashGit I commented on the created issue that the attachment is empty. Table navigation is expected to be done using arrow keys. Are those buttons in the table not accessible with arrow keys?

**Comment by BhanuPrakashGit** [comment]
GithubTags:#Regressed:01-15-24;#Closed;
Verified the bug in the environment:
Version: 1.48.0-insider (Universal)
Commit: https://github.com/microsoft/azuredatastudio/commit/e5e1204d4fb4d4e880cdd893347752ee5736459a
Date: 2023-12-05T06:38:01.191Z
VS Code: 1.82.0
Electron: 25.9.6
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.2.0

Issue is fixed, we are able to navigate over the button with the help of the arrow keys and voiceover announcing it as button. Hence closing the bug.
PFA for reference

[25150_FocusIssue.zip](https://github.com/microsoft/azuredatastudio/files/13937216/25150_FocusIssue.zip)
